web-console/3-project-details/labels.md

```diff
@@ -0,0 +1,104 @@
+
+## Edit Labels
+![Label 1](img/new-labels-01-edit.png)
+
+- The same `Manage Labels` link will be used for editing existing labels.
+- Users can change a name or value as desired, as long as the name is not already associated with another value.
+
+#### Implementation Details
+- Values can be edited in this view, but only the current object will receive the label change. In order to bulk change labels for multiple items (Deployments in this case), users must go to the summary page and multi-select items.
+- If there are any restricted labels that users are unable to change, the entire row should be disabled.
+
+## Remove Labels
+![Label 1](img/new-labels-01-delete.png)
+
+- The same `Manage Labels` link will be used for removing existing labels.
+- Clicking the x icon next to a label will remove that row and enable the save button.
+- Applying these changes will only remove the label from this specific object, not from the system entirely.
+
+![Label 1](img/new-labels-02-delete.png)
+
+# Labels on Summary Pages
+
+![Label 1](img/new-labels-01-manage.png)
+
+- Summary lists are provided for each major section such as Deployments, Pods, Builds, Images, Storage, etc.
+- Currently users can filter those lists based on their applied labels, in order to narrow the results.
+- Filtering by label should remain, but checkboxes should be added to the left side of each row to allow for multi-select and bulk actions from the summary page.
+
+## Manage Labels
+- Users can multi-select using the checkboxes provided and a `Manage Labels` button should be available in the top right corner of the page.
+- The `Manage Labels` button will open a modal for any add, edit, or remove actions to take place.
+  - The top of the modal should summarize the selections to reiterate the number of objects that will be affected by the label changes. Intro text detailed below:
+  - "You have selected 'x' 'Object-Type.' The following changes will be applied to all selections.
+
+### Add
+![Label 1](img/new-labels-02-manage.png)
+
+- Users can add a single label or multiple labels to all selected objects.
+- If the selected objects have any labels in common, the labels will be listed in rows above the empty one. If not, an empty row will be presented alone, with the option to add more labels.
+- Adding the new label and clicking `Save` will close the modal and initiate a toast notification to confirm the changes have been applied to all objects.
+
+![Label 1](img/new-labels-02C-manage.png)
+- If users select objects without first filtering for common labels, some changes may overwrite existing label values.
+- When this is the case, clicking save will replace the modal contents with a warning and confirmation message to inform users that the label change will overwrite an existing label value.
+- Users must confirm the changes by clicking `Apply` before changes will be saved.
```

Comment:
Your inline notification says "Saving these changes..." but the action is Apply. Maybe it makes sense to reverse the buttons: Apply is the first action, and if there's a conflict then the button is Save. That would match the text. Alternatively, you could change the inline notification to say "Applying these changes..."

Reply:
Yeah I've been debating that same thing @ncameronbritt! 

@serenamarie125 @cshinn any preference on the primary action for the "Manage Labels" modal? We are torn between "Apply" and "Save."

Reply:
I don't have a strong preference, but the action should match whatever we tell the users in the notification.

Reply:
Yeah agreed, I'll adjust that once I hear back from others :+1: